spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/ServiceInstanceListSupplierBuilder.java

```diff
@@ -301,6 +302,16 @@ public ServiceInstanceListSupplierBuilder withHints() {
 		return this;
 	}
 
+	public ServiceInstanceListSupplierBuilder withSubset() {
+		DelegateCreator creator = (context, delegate) -> {
+			PropertyResolver resolver = context.getBean(PropertyResolver.class);
+			LoadBalancerClientFactory factory = context.getBean(LoadBalancerClientFactory.class);
+			return new SubsetServiceInstanceListSupplier(delegate, resolver, factory);
+		};
+		creators.add(creator);
+		return this;
+	}
+
```

Comment:
Please also add  `configurations` property-based instantiation in `LoadBalancerClientConfiguration`.